lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,21 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          return false if ends_with_equals? sym_name
+          valid_19_syntax_symbol? sym_name
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
+          symbol = sym_name.sub(':', '')
+          if symbol.size == 1
+            symbol =~ /\A[a-zA-Z_]/
+          else
+            symbol =~ /\A[a-zA-Z_][a-zA-Z0-9]/
+          end
+        end
```

Comment:
This implementation will be falsy for `:__foo__` and `:łódź` (both valid symbols) and truthy for `:foo:bar` (an invalid symbol, but one that wouldn’t even parse in the first place, so probably would blow up earlier).